### PR TITLE
[supervisor] Persist bash history across workspace restarts PRD-59

### PR DESCRIPTION
**Pull request by iQQBot**
## Related Issue(s)

Relates [PRD-59](https://linear.app/gitpod/issue/PRD-59/persisted-terminal-history)

## How to test

1. start a workspace in your preview environment
2. execute some command in this workspace
3. restart this workspace
4. use up arrow key to check if history still in there

repeat these step in different ide, i.e. Jetbrains, ssh, vscode desktop

#### Preview status

<p>Gitpod was successfully deployed to your preview environment.</p>
<ul>
	<li><b>🏷️ Name</b> - pd-bash-history</li>
	<li><b>🔗 URL</b> - <a href="https://pd-bash-history.preview.gitpod-dev.com/workspaces" target="_blank">pd-bash-history.preview.gitpod-dev.com/workspaces</a>.</li>
	<li><b>📚 Documentation</b> - See our <a href="https://www.notion.so/gitpod/6debd359591b43688b52f76329d04010#7c1ce80ab31a41e29eff2735e38eec39" target="_blank">internal documentation</a> for information on how to interact with your preview environment.</li>
</ul>

## Build Options:

- [ ] /werft with-werft
      Run the build with werft instead of GHA
- [ ] leeway-no-cache
- [ ] /werft no-test
      Run Leeway with `--dont-test`

<details>
<summary>Publish Options</summary>

- [ ] /werft publish-to-npm
- [ ] /werft publish-to-jb-marketplace
</details>

<details>
<summary>Installer Options</summary>

- [ ] analytics=segment
- [ ] with-dedicated-emulation
- [x] with-ws-manager-mk2
- [ ] workspace-feature-flags
  Add desired feature flags to the end of the line above, space separated
</details>

#### Preview Environment Options:
- [ ] /werft with-local-preview
      If enabled this will build `install/preview`
- [x] /werft with-preview
- [ ] /werft with-large-vm
- [x] /werft with-gce-vm
      If enabled this will create the environment on GCE infra
- [ ] with-integration-tests=all
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`

/hold

### CODE REVIEW

File: components/supervisor/pkg/supervisor/supervisor.go [hunk 1]
```diff
@@ -1006,6 +1006,9 @@ func buildChildProcEnv(cfg *Config, envvars []string, runGP bool) []string {
 		envs["JAVA_TOOL_OPTIONS"] += fmt.Sprintf(" -Xmx%sm", mem)
 	}
 
+	envs["HISTFILE"] = "/workspace/.gitpod/.bash_history"
```
Comment (mustard-mh) [inline components/supervisor/pkg/supervisor/supervisor.go:1009]: Should we check if this two env already exists?

[the remark this answers is not in the document]
Comment (akosyakov) [reply components/supervisor/pkg/supervisor/supervisor.go:1009]: or maybe we just rename it to `.shell_history`?

it seems zsh is using the same env var?

**Comment by akosyakov** [comment]
I'm not sure it is actually fixing PRD-59. It is definitely an improvement, but as far as I remember the original issue was about preserving history for some context, not only workspace. See https://github.com/gitpod-io/gitpod/issues/9019#issuecomment-1182900593 for instance. cc @loujaybee 

I think we should merge it though.

**Comment by laushinka** [comment]
> I'm not sure it is actually fixing PRD-59. It is definitely an improvement, but as far as I remember the original issue was about preserving history for some context, not only workspace. 

Exactly, so I updated the description to `Relates`. @loujaybee will update the scope in [PRD-59](https://linear.app/gitpod/issue/PRD-59/persisted-terminal-history). 

💯 that this is definitely an improvement. Thanks for getting on this very quickly, @iQQBot!

**Comment by filiptronicek** [comment]
@iQQBot would a title like "Persist bash history across workspace restarts" work better? Not sure if the title is too important, but just throwing it out there.

**Review by akosyakov — APPROVED** [review]
I have not tried but look sensible.

please unhold when you are ready

/hold

**Pull request merged**